File: LABORATORIO_8/PREG01_LAB08-2024-2/Biblioteca/PlatoFondo.cpp
//
// Created by USER on 18/06/2026.
//

#include "PlatoFondo.h"

PlatoFondo::PlatoFondo() {
    proteina=0;
}

PlatoFondo:: ~PlatoFondo() {

}


int PlatoFondo::get_proteina() const {
    return proteina;
}

void PlatoFondo::set_proteina(int proteina) {
    this->proteina = proteina;
}

void PlatoFondo::lee(ifstream &arch) {
    Producto::lee(arch);
    arch>>proteina;
}

void PlatoFondo::imprime(ofstream &arch) {
    Producto::imprime(arch);
    if (proteina==1)
        arch<<setw(5)<<" "<<"Proteina: Pollo"<<endl;
    if (proteina==2)
        arch<<setw(5)<<" "<<"Proteina: Carne"<<endl;
    if (proteina==3)
        arch<<setw(5)<<" "<<"Proteina: Pescado"<<endl;
    if (proteina==4)
        arch<<setw(5)<<" "<<"Proteina: Lacteos"<<endl;

}

File: Restaurante_2024-2/src/PlatoFondo.java
import java.util.Scanner;

public class PlatoFondo extends Producto{
    private int proteina;

    public PlatoFondo(){
        proteina=0;
    }

    public int getProteina() {
        return proteina;
    }

    public void setProteina(int proteina) {
        this.proteina = proteina;
    }

    //142 E Bruschetta 4 picante
    //142 P Pasta_Alfredo 10 4
    //142 P Pollo_a_la_Brasa 15 1
    @Override
    public String tiposoy(){
        return "Plato Fondo";
    }

    @Override
    public void lee(Scanner arch){
        super.lee(arch);
        proteina = arch.nextInt();

    }

    @Override
    public void imprime(){
        super.imprime();
        if(proteina==1)
            System.out.println("Pollo");
        if(proteina==2)
            System.out.println("Carne");
        if(proteina==3)
            System.out.println("Pescados");
        if(proteina==4)
            System.out.println("Lacteos");

    }

}
